### PR TITLE
Improve efficiency of ragged batching scheduler

### PR DESCRIPTION
**Pull request by tohtana**
This PR includes the following optimization in the ragged batching scheduler:
- Remove inefficient loop in TopP logits processor (390302476251c9d58ea443900a0388e8d006f19f): The loop in Top P processor is inefficient when it runs on multiple requests. This PR replaces the loop with a batch API with indexing. 
- Copying `is_done` flag of requests to host memory as a batch (678673734503b14ed5a9b9dd7ec7d7e1efa6d0eb): is_done is computed on cuda but implicitly copied to host every time it is evaluated as bool. This PR copies it to host at once and stores them as Python bool values.

### CODE REVIEW

**Comment by tohtana** [comment]
Closing because #377 is an improved version of this PR.
